### PR TITLE
Adding grinder scripts for exercising annotations ingest

### PR DESCRIPTION
**Pull request by tilogaat**

### CODE REVIEW

File: contrib/grinder/properties/grinder.properties [hunk 1]
```diff
@@ -1,18 +1,21 @@
-grinder.script=../scripts/tests.py
```
Comment (GeorgeJahad) [inline contrib/grinder/properties/grinder.properties]: this file is meant for tests.  i like the idea of creating another one for localhost, but that should be a separate properties file, (and it should be mentioned in the readme.)

File: contrib/grinder/scripts/annotationsingest.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    The metrics are a list of batches.  Each batch is a list of metrics processed by
+    a single metrics ingest request.
+    """
+    annotations =  generate_metrics_tenants(default_config['num_tenants'],
+                                            default_config['annotations_per_tenant'], agent_number,
+                                            default_config['num_nodes'], 
+                                            cls.generate_annotations_for_tenant)
+
+    cls.annotations = cls.divide_annotations_into_batches(annotations, 1)
```
Comment (GeorgeJahad) [inline contrib/grinder/scripts/annotationsingest.py]: This is a bit wierd, creating a batch size of 1.  The batches serve no purpose in this module and should be eliminated.

Comment (tilogaat) [inline contrib/grinder/scripts/annotationsingest.py]: fixed 35ca585

File: contrib/grinder/properties/grinder.properties (hunk 1 above)
Comment (tilogaat) [inline contrib/grinder/properties/grinder.properties]: Fixed 35ca585

File: contrib/grinder/properties/grinder.properties.local [hunk 3]
```diff
@@ -0,0 +1,23 @@
+grinder.script=../scripts/tests.py
```
Comment (GeorgeJahad) [inline contrib/grinder/properties/grinder.properties.local]: I thought the purpose of the ".local" file was to run the grinder, not the tests.  Shouldn't this be grinder.py?

File: contrib/grinder/scripts/annotationsingest.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            'tags': 'tag',
+            'data': 'data'}
+
+  def generate_payload(self, time, batch):
+    payload = self.generate_annotation(time,batch[1])
+    return json.dumps(payload)
+
+  def ingest_url(self):
+    return "%s/v2.0/tenantId/events" % default_config['url']
+
```
Comment (GeorgeJahad) [inline contrib/grinder/scripts/annotationsingest.py]: It looks like you are ingesting to the same tenant every time.  Wouldn't it be better to distribute across tenants by using the tenantId, created in generate_annotations_for_tenant() method above?

File: contrib/grinder/scripts/annotationsingest.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  request = HTTPRequest()
+  test1.record(request)
+
+
+  @classmethod
+  def create_metrics(cls, agent_number):
+    """ Generate all the metrics for this worker
+
+    The metrics are a list of batches.  Each batch is a list of metrics processed by
+    a single metrics ingest request.
```
Comment (GeorgeJahad) [inline contrib/grinder/scripts/annotationsingest.py]: This comment should be updated now that batches are removed.

File: contrib/grinder/scripts/grinder.py [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
 import ingest
+import annotationsingest
 import query
 
 #ENTRY POINT into the Grinder
 
 #The code inside the TestRunner class is gets executed by each worker thread
 #Outside the class is executed before any of the workers begin
-thread_manager = ingest.ThreadManager(grinder)
+thread_manager = annotationsingest.ThreadManager(grinder)
```
Comment (GeorgeJahad) [inline contrib/grinder/scripts/grinder.py]: This change is actually unnecessary, and should probably be removed.

Comment (tilogaat) [inline contrib/grinder/scripts/grinder.py]: @GeorgeJahad I could not get it to work without this change.

File: contrib/grinder/scripts/tests.py [hunk 7]
```diff
@@ -87,12 +90,14 @@ def test_init_process(self):
     self.assertEqual(type(t1), query.QueryThread)
```
Comment (GeorgeJahad) [inline contrib/grinder/scripts/tests.py]: I'd like a test here confirming that the first thread after the ingest and query threads is an annotation thread.

File: contrib/grinder/scripts/tests.py [hunk 8]
```diff
@@ -87,12 +90,14 @@ def test_init_process(self):
...
-    tot_threads = (ingest.default_config['ingest_concurrency'] + ingest.default_config['query_concurrency'])
+    tot_threads = (ingest.default_config['ingest_concurrency'] + ingest.default_config['query_concurrency'] + ingest.default_config['annotations_concurrency'])
     self.assertRaises(Exception,self.tm.setup_thread, tot_threads)
 
-
     #confirm that the correct batches of ingest metrics are created for worker 0
     self.tm.create_all_metrics(0)
+    self.assertEqual(annotationsingest.AnnotationsIngestThread.annotations,
+                             [[0, 0], [0, 1], [1, 0], [1, 1]])
+
```
Comment (GeorgeJahad) [inline contrib/grinder/scripts/tests.py]: Add tests for the slice's like the ingest and query slice tests below.

File: contrib/grinder/properties/grinder.properties.local (hunk 3 above)
Comment (tilogaat) [inline contrib/grinder/properties/grinder.properties.local]: Fixed 6262254

File: contrib/grinder/scripts/annotationsingest.py (hunk 4 above)
Comment (tilogaat) [inline contrib/grinder/scripts/annotationsingest.py]: Fixed in 8795308

File: contrib/grinder/scripts/annotationsingest.py (hunk 5 above)
Comment (tilogaat) [inline contrib/grinder/scripts/annotationsingest.py]: Fixed in 8795308

File: contrib/grinder/scripts/tests.py (hunk 8 above)
Comment (tilogaat) [inline contrib/grinder/scripts/tests.py]: 2bb5838

File: contrib/grinder/scripts/tests.py (hunk 7 above)
Comment (tilogaat) [inline contrib/grinder/scripts/tests.py]: 2bb5838

**Comment by GeorgeJahad** [comment]
+1

**Pull request merged**